### DISCUSSION TITLE
Best practice for updating a status field after a time interval

### DISCUSSION THREAD

**Discussion by MotaZor94** [supabase/supabase#14800] [Questions] [answered]
Hi Supabase Community!

I'm working on an application where I have a 'listings' table, and each listing has an 'expiration_date' and a 'status'. Once a payment is successful, the listing's expiration_date is set to 30 days in the future, and the status is set to 'active'.

However, I'm not quite sure about the best way to handle the status update once the listing has expired. After the expiration_date passes, I'd like to automatically update the status to 'expired'.

Would you be able to give me some examples on what to do in this case?

Should I run a cron job, or is there some built in functionality for this kind of cases?

Thanks!

**Comment by GaryAustin1** [comment, marked as answer]
Run a cron job and then update all rows past the required date.
